src/OrbitGl/CallstackThreadBar.cpp

```diff
@@ -175,10 +175,26 @@ void CallstackThreadBar::OnPick(int x, int y) {
 }
 
 void CallstackThreadBar::SelectCallstacks() {
-  Vec2& from = mouse_pos_last_click_;
-  Vec2& to = mouse_pos_cur_;
+  Vec2 from = mouse_pos_last_click_;
+  Vec2 to = mouse_pos_cur_;
 
-  time_graph_->SelectCallstacks(from[0], to[0], GetThreadId());
+  if (from > to) {
+    std::swap(from, to);
+  }
+
+  uint64_t t0 = time_graph_->GetTickFromWorld(from[0]);
+  uint64_t t1 = time_graph_->GetTickFromWorld(to[0]);
+
+  int64_t thread_id = GetThreadId();
+  bool thread_id_is_all_threads = thread_id == orbit_base::kAllProcessThreadsTid;
+
+  CHECK(capture_data_);
+  auto& selected_callstack_events =
```

Comment:
I am thinking of wheather we want to move this logic into the "SelectCallstackEvents" method. DataManager should also have access to capture data.

Reply:
As discussed, there is no access to capture data from the data manager. I'll leave this here.